res/res_rtp_asterisk.c

```diff
@@ -4257,12 +4259,30 @@ static int ast_rtp_dtmf_begin(struct ast_rtp_instance *instance, char digit)
 		return -1;
 	}
 
-	/* Grab the payload that they expect the RFC2833 packet to be received in */
-	payload = ast_rtp_codecs_payload_code_tx(ast_rtp_instance_get_codecs(instance), 0, NULL, AST_RTP_DTMF);
+	if (rtp->lasttxformat == ast_format_none) {
+		/* we haven't sent any audio yet so we have to lookup both the preferred payload type and
+		   bitrate. */
+		payload_format = ast_rtp_codecs_get_preferred(ast_rtp_instance_get_codecs(instance));
```

Comment:
This should handle preferred not being present, if it isn't then fall back to 8000